pints/_transformation.py

```diff
@@ -0,0 +1,1120 @@
+        transform = pints.RectangularBoundariesTransformation(boundaries)
+
+    Extends :class:`ElementWiseTransformation`.
+    """
+    def __init__(self, lower_or_boundaries, upper=None):
+        # Parse input arguments
+        if upper is None:
+            if not isinstance(lower_or_boundaries,
+                              pints.RectangularBoundaries):
+                raise ValueError(
+                    'RectangularBoundariesTransformation requires a lower and '
+                    'an upper bound, or a single RectangularBoundaries object.'
+                )
+            boundaries = lower_or_boundaries
+        else:
+            # Create RectangularBoundaries for all the input checks
+            boundaries = pints.RectangularBoundaries(lower_or_boundaries,
+                                                     upper)
+
+        self._a = boundaries.lower()
+        self._b = boundaries.upper()
+
+        # Cache dimension
+        self._n_parameters = boundaries.n_parameters()
+        del(boundaries)
+
+    def jacobian(self, q):
+        """ See :meth:`Transformation.jacobian()`. """
+        q = pints.vector(q)
+        diag = (self._b - self._a) / (np.exp(q) * (1. + np.exp(-q)) ** 2)
+        return np.diag(diag)
+
+    def jacobian_S1(self, q):
+        """ See :meth:`Transformation.jacobian_S1()`. """
+        q = pints.vector(q)
+        n = self._n_parameters
+        jac = self.jacobian(q)
+        jac_S1 = np.zeros((n, n, n))
+        rn = np.arange(n)
+        jac_S1[rn, rn, rn] = np.diagonal(jac) * (np.exp(-q) - 1.) * expit(q)
+        return jac, jac_S1
+
+    def log_jacobian_det(self, q):
+        """ See :meth:`Transformation.log_jacobian_det()`. """
+        q = pints.vector(q)
+        s = self._softplus(-q)
+        return np.sum(np.log(self._b - self._a) - 2. * s - q)
+
+    def log_jacobian_det_S1(self, q):
+        """ See :meth:`Transformation.log_jacobian_det_S1()`. """
+        q = pints.vector(q)
+        logjacdet = self.log_jacobian_det(q)
+        dlogjacdet = 2. * np.exp(-q) * expit(q) - 1.
+        return logjacdet, dlogjacdet
+
+    def n_parameters(self):
+        """ See :meth:`Transformation.n_parameters()`. """
+        return self._n_parameters
+
+    def multiple_to_model(self, qs):
+        """ See :meth:`Transformation.multiple_to_model()`. """
+        qs = np.asarray(qs)
+        return (self._b - self._a) * expit(qs) + self._a
+
+    def to_model(self, q):
+        """ See :meth:`Transformation.to_model()`. """
+        q = pints.vector(q)
+        return (self._b - self._a) * expit(q) + self._a
+
+    def to_search(self, p):
+        p = pints.vector(p)
+        """ See :meth:`Transformation.to_search()`. """
+        return np.log(p - self._a) - np.log(self._b - p)
+
+    def _softplus(self, q):
+        """ Returns the softplus function. """
+        return np.log(1. + np.exp(q))
+
+
+class ScalingTransformation(ElementWiseTransformation):
```

Comment:
Thanks @chonlei -- looks great! This is exactly the transformation I was wanting to have in there...

Reply:
Great!